src/main/java/com/mraof/minestuck/entity/underling/UnderlingEntity.java

```diff
@@ -174,8 +173,15 @@ protected void onDeathUpdate()
 			{
 				for(GristAmount gristAmount : grist.getAmounts())
 				{
-					if(gristAmount.getAmount() > 0)
-						this.world.addEntity(new GristEntity(world, randX(), this.getPosY(), randZ(), gristAmount));
+					long gristLeft = gristAmount.getAmount();
+
+					while(gristLeft > (gristAmount.getAmount() / (int) ((Math.random() * 10) + 1)) && (long)(gristLeft*0.9) > 0) {
+
+								this.world.addEntity(new GristEntity(world, randX(), this.getPosY(), randZ(), new GristAmount(gristAmount.getType(), (long) (gristLeft * 0.9))));
+								gristLeft = gristLeft - (long)((gristLeft*0.9)+1);
```

Comment:
Why is there a +1 here? I'd assume that you would only remove what was dropped (gristLeft * 0.9)